compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangUserDefinedType.java

```diff
@@ -71,9 +71,18 @@ public void accept(BLangNodeVisitor visitor) {
 
     @Override
     public String toString() {
+        String typeName = this.typeName.value;
+
+        if (typeName.startsWith("$") && this.type != null) {
```

Comment:
Can we check for the anon flag instead?

Reply:
Since this is `toString` method I think we need to make sure we can call this method at any point in compiler phases.

Reply:
No, the type is `SymbolTable.noType` (and symbol in null) since I don't have access to an instance of that class, only way I can check for that is to use `toString` and see if it's `other`.

Reply:
Can't we check the type tag?